src/IpcEventDispatcher.h

```diff
@@ -0,0 +1,40 @@
+// Copyright 2015, Christopher J. Foster and the other displaz contributors.
+// Use of this code is governed by the BSD-style license found in LICENSE.txt
+
+#ifndef DISPLAZ_IPCED_H_INCLUDED
+#define DISPLAZ_IPCED_H_INCLUDED
+
+#include <QByteArray>
+#include <QKeySequence>
+#include <QShortcut>
+
+class PointViewerMainWindow;
+
+/// Class for dispatching hook events and payloads to given IpcChannel
+///
+class IpcEventDispatcher : public QObject
+{
+    Q_OBJECT
+    public:
+        IpcEventDispatcher(QObject* parent = NULL, PointViewerMainWindow* getPayload = NULL,
+                           QByteArray = 0, QByteArray = 0, int = 0);
```

Comment:
It's probably not a good idea to default all these arguments, because if some of them are ever null (particularly `getPayload`) you're likely to get a nice segfault :-)